### PR TITLE
[s3inbox] do authentication before we start processing the request

### LINKED ISSUE #655: [s3inbox] authenticate before handling request
**Describe the bug**
Currently we start processing the incoming request before we check if the user is allowed to use the service.
This leads to he user getting errors like 403 or 406 when in reality the reported error should have been 401.

**Steps to reproduce**

**Expected behavior**
First we need to check that the user has a valid token and are allowed to access the service.
Next we check that the request is of a type that we allow.

**Additional context**

**Estimation of size**: small/medium/big

**Estimation of priority**: low/medium/high

### PR DESCRIPTION
**Pull request by jbygdell**
**Description**
As per title.

closes #655

### CODE REVIEW

**Review by pontus — DISMISSED** [review]
👍

**Review by pahatz — APPROVED** [review]
Looks good!

**Review by MalinAhlberg — APPROVED** [review]
Good good

File: sda/internal/userauth/userauth.go [hunk 1]
```diff
@@ -58,27 +58,6 @@ func (u *ValidateFromToken) Authenticate(r *http.Request) (jwt.Token, error) {
 			return nil, fmt.Errorf("failed to get issuer from token (%v)", iss)
 		}
 
-		// Check whether token username and filepath match
```
[the remark this answers is not in the document]
Comment (jbygdell) [reply sda/internal/userauth/userauth.go:61]: Because the users are supposed to be chrooted, this is the way we enforce that for S3

**Pull request merged**